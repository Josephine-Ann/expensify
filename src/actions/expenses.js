import uuid from 'uuid';


export const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {} ) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',
    id  
})


// REMOVE_EXPENSE
// EDIT_EXPENSE

// figure out if it's a string? 
// righteeooooh, vale, a veure...

export const editExpense = ( id, updates ) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates  
})
