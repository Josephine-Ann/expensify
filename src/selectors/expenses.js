export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate;
        const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate;
        // code above...when we start, we won't be strings. if we have not been set,
        // please...set us!
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt > b.createdAt ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount > b.amount ? 1 : -1;
        }
    });
};

