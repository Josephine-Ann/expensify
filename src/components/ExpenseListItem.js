import React from 'react'; 
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, description, amount, createdAt, id }) => (
    <div>
    <p>{description}</p>
    <p>{amount}</p>
    <p>{createdAt}</p>
    <button onClick={() => {
        dispatch(removeExpense(id));
    }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);

// import React from 'react';
// import { connect } from 'react-redux';
// import { setTextFilter } from '../actions/filters';

// const ExpenseListFilters = (props) => (
//     <div>
//     <input type="text" value={props.filters.text} onChange={(e) => {
//         props.dispatch(setTextFilter(e.target.value));
//         console.log(e.target.value)
//     }}/>
//     </div>
// );

// const mapStateToProps = (state) => {
//     return {
//         filters: state.filters
//     }
// }

// export default connect(mapStateToProps)(ExpenseListFilters);
