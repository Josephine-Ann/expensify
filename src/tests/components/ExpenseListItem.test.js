import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';


test('should render ExpenseListItem with expense', () => {
    const expense = expenses[0]
    const wrapper = shallow(<ExpenseListItem key={expense.id} description={expense.description} createdAt={expense.createdAt} amount={expense.amount}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should render ExpenseListItem with expense', () => {
    const wrapper = shallow(<ExpenseListItem key={[]} description={[]} createdAt={[]} amount={[]}/>);
    expect(wrapper).toMatchSnapshot();
});