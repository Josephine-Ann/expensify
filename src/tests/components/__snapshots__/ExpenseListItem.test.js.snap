// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expense 1`] = `
<div>
  <Link
    to="/edit/undefined"
  >
    <p>
      Gum
    </p>
  </Link>
  <p>
    $1.95
    -
    January 1st, 1970
  </p>
</div>
`;

exports[`should render ExpenseListItem with expense 2`] = `
<div>
  <Link
    to="/edit/undefined"
  >
    <p />
  </Link>
  <p>
    $0.00
    -
    May 5th, 2020
  </p>
</div>
`;
